Server/Test_Plan_Files_Antivirus.md

```diff
@@ -0,0 +1,152 @@
+
+### clamAV executable mode upload
+
+1. setup the file firewall app to use clamAV in "Executable" mode
+2. upload `eicar.com`, `eicar_com.zip` and `eicarcom2.zip` from http://www.eicar.org/85-0-Download.html via all available endpoints. (See "Tips how to test different endpoints with curl")
+
+None of this uploads should be possible.
+
+| upload test ↓ Testfile→       | eicar.com | eicar_com.zip | eicarcom2.zip |
+| ----------------------------- | --------- | ------------- | ------------- |
+| old dav path without chunking	|    🚧     |       🚧       |      🚧       |		
+| old dav path with chunking    |    🚧     |       🚧       |      🚧       |
+| new dav path without chunking |    🚧     |       🚧       |      🚧       |
+| new dav path with chunking    |    🚧     |       🚧       |      🚧       |
+| public upload                 |    🚧     |       🚧       |      🚧       |
+| files_texteditor app          |    🚧     |       -       |      -        |
+
+### clamAV daemon mode upload
+
+1. setup the file firewall app to use clamAV in "Daemon" mode
+2. upload `eicar.com`, `eicar_com.zip` and `eicarcom2.zip` from http://www.eicar.org/85-0-Download.html via all available endpoints. (See "Tips how to test different endpoints with curl")
+
+None of this uploads should be possible.
+
+| upload test ↓ Testfile→       | eicar.com | eicar_com.zip | eicarcom2.zip |
+| ----------------------------- | --------- | ------------- | ------------- |
+| old dav path without chunking	|    🚧     |       🚧       |      🚧       |		
+| old dav path with chunking    |    🚧     |       🚧       |      🚧       |
+| new dav path without chunking |    🚧     |       🚧       |      🚧       |
+| new dav path with chunking    |    🚧     |       🚧       |      🚧       |
+| public upload                 |    🚧     |       🚧       |      🚧       |
+| files_texteditor app          |    🚧     |       -       |      -        |
+
+### clamAV daemon (socket) mode upload
+
+1. setup the file firewall app to use clamAV in "Daemon (Socket)" mode
+2. upload `eicar.com`, `eicar_com.zip` and `eicarcom2.zip` from http://www.eicar.org/85-0-Download.html via all available endpoints. (See "Tips how to test different endpoints with curl")
+
+None of this uploads should be possible.
+
+| upload test ↓ Testfile→       | eicar.com | eicar_com.zip | eicarcom2.zip |
+| ----------------------------- | --------- | ------------- | ------------- |
+| old dav path without chunking	|    🚧     |       🚧       |      🚧       |		
+| old dav path with chunking    |    🚧     |       🚧       |      🚧       |
+| new dav path without chunking |    🚧     |       🚧       |      🚧       |
+| new dav path with chunking    |    🚧     |       🚧       |      🚧       |
+| public upload                 |    🚧     |       🚧       |      🚧       |
+| files_texteditor app          |    🚧     |       -       |      -        |
+
+### Filesize limit
+
+set "File size limit" to 100 bytes
```

Comment:
does this set the maximum size file to bother scanning?
i.e. the small virus gets blocked, and the big virus is allowed through

Reply:
I think this setting can be used for performance reasons, e.g. not scan GB size video files

Reply:
OK, then the test is the correct way around.